### PR TITLE
Add recipe list views

### PR DESCRIPTION
**Pull request by seowalex**
# TODO

- [ ] ~Figure out how to get sidebar selection to persist~
- [x] Filter by recipes by ingredients
- [x] Ingredient list view
- [x] Grid view?

### CODE REVIEW

**Review by sc-arecrow — APPROVED** [review]
LGTM

File: ChopChop/Storage/Models/QuantityRecord.swift [hunk 1]
```diff
@@ -9,6 +9,17 @@ extension QuantityRecord: Codable {
         case count, mass, volume, unit
     }
 
+    var type: BaseQuantityType {
+        switch self {
+        case .count:
+            return .count
+        case .mass(_, unit: _):
```
Comment (sc-arecrow) [inline ChopChop/Storage/Models/QuantityRecord.swift]: I think the `(_, unit: _)` part might be unnecessary

File: ChopChop/Views/Ingredients/IngredientCollectionView.swift [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        }
+        .onDisappear {
+            viewModel.query = ""
+            viewModel.filterByExpiryDate = false
+            viewModel.expiryDateStart = .today
+            viewModel.expiryDateEnd = .today
+        }
+    }
+
+    func ExpiryDatePicker() -> some View {
```
Comment (sc-arecrow) [inline ChopChop/Views/Ingredients/IngredientCollectionView.swift]: Any reason why this and the subsequent views with no arguments are functions and not computed properties? similar to `body` like
```
var expiryDatePicker: some View {}
```

File: ChopChop/Views/Recipes/RecipeCollectionView.swift [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                }
+            }
+        }
+        .onDisappear {
+            viewModel.query = ""
+            viewModel.selectedIngredients.removeAll()
+        }
+    }
+
+    func NotFoundView() -> some View {
```
Comment (sc-arecrow) [inline ChopChop/Views/Recipes/RecipeCollectionView.swift]: Same as `IngredientCollectionView`

File: ChopChop/Views/Sidebar.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            NavigationLink(
+                destination: RecipeCollectionView(viewModel: RecipeCollectionViewModel(title: "Uncategorised"))
+            ) {
+                Image(systemName: "questionmark.folder")
+                Text("Uncategorised")
+            }
+        }
+    }
+
+    func IngredientsSection() -> some View {
```
Comment (sc-arecrow) [inline ChopChop/Views/Sidebar.swift]: Same as `IngredientCollectionView`

**Review by sc-arecrow — APPROVED** [review]
LGTM other than the comments I raised

File: ChopChop/Storage/Models/QuantityRecord.swift (hunk 1 above)
Comment (seowalex) [reply ChopChop/Storage/Models/QuantityRecord.swift]: Ah yes will fix.

File: ChopChop/Views/Ingredients/IngredientCollectionView.swift (hunk 2 above)
Comment (seowalex) [reply ChopChop/Views/Ingredients/IngredientCollectionView.swift]: A good point, I'll change them.

File: ChopChop/Views/Recipes/RecipeCollectionView.swift (hunk 3 above)
Comment (seowalex) [reply ChopChop/Views/Recipes/RecipeCollectionView.swift]: Noted.

File: ChopChop/Views/Sidebar.swift (hunk 4 above)
Comment (seowalex) [reply ChopChop/Views/Sidebar.swift]: Noted.

**Pull request merged**